pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -196,7 +199,7 @@ func (ssr *Snapshotter) Run(stopCh <-chan struct{}, startWithFullSnapshot bool)
 			return fmt.Errorf("failed to reset full snapshot timer: %v", err)
 		}
 	}
-
+	go ssr.RenewFullSnapshotLeasePeriodically(fullSnapshotLeaseUpdateRetryInterval)
```

Comment:
Placing a go routine here cause many problems in following cases: 
1. When snapshotter itself is not enabled ? In `etcd-events` we don't enable the snapshottter.
2. When Snapshot Lease Renewal is not enabled 
3. How to make sure this go routine will graceful terminates ?

Reply:
> When snapshotter itself is not enabled ? In etcd-events we don't enable the snapshottter.

Then the goroutine will also not start, it is run only when the snapshotter starts and ends when the snapshotter closes.

> When Snapshot Lease Renewal is not enabled

If renewal is not enabled, then there won't be any attempts to update the lease, but the timer is called only first time, then after it is not called and goroutine just sits there idle. So no affect on leases. 

> How to make sure this go routine will graceful terminates ?

This is terminated with the help of a stop channel. When the snapshotter stops ( due to any reason ), then as part of the cleanup, a message is sent on the `FullSnapshotLeaseStopCh`. 

**The life of the goroutine is strictly tied to life of the snapshotter**. And we want the lease to be updated while the snapshotter is taking snapshots right? Hence we also want the goroutine to run. That's why goroutine is run when the snapshotter starts, and ends when snapshotter ends. 

Pls correct me if I'm wrong

Reply:
> Then the goroutine will also not start, it is run only when the snapshotter starts and ends when the snapshotter closes

make sense

> If renewal is not enabled, as from the code below, then there won't be any attempts to update the lease
case <-ssr.FullSnapshotLeaseUpdateTimer.C:
	if ssr.healthConfig.SnapshotLeaseRenewalEnabled {

I'm still not agreeing as we are starting go routine for doing nothing (in case lease renewal is not enabled), then why not check condition this before starting a go routine, WDYT ?

Reply:
> why not check condition this before starting a go routine, WDYT ?

Yea that would be good. I'll address this in the next commit. Thanks!

Reply:
Updated it in the latest [commit](https://github.com/gardener/etcd-backup-restore/pull/711/commits/fd879743025e89517a9bcec447358da7d3e97f97). PTAL